packages/react-native-hid/src/index.js

```diff
@@ -18,20 +46,27 @@ export default class HIDTransport extends Transport<DeviceObj> {
     return list.filter(d => d.vendorId === ledgerUSBVendorId);
   }
 
-  static listen(observer: *) {
-    if (!NativeModules.HID) return { unsubscribe: () => {} };
-    let unsubscribed = false;
+  static listDevices(){
     HIDTransport.list().then(candidates => {
       for (const c of candidates) {
-        if (!unsubscribed) {
+        if (!HIDTransport.unsubscribed) {
           const deviceModel = identifyUSBProductId(c.productId);
-          observer.next({ type: "add", descriptor: c, deviceModel });
+          HIDTransport.observer && HIDTransport.observer.next({ type: "add", descriptor: c, deviceModel });
         }
       }
     });
+  }
+
+  static listen(observer: *) {
+    HIDTransport.observer = observer;
```

Comment:
Please add a failsafe that throw an error if `HIDTransport.observer` is still not `null` . otherwise we will have ugly race condition since the code provide here don't allow to run two listen at same time.

warning: we maybe have a problem in the current mobile app if you open the Receive flow when also being in Manager? 🤔

**Here is what I suggest:**
- after you add this failsafe (once at a time), If there is no problem with current mobile app. fine, let's ship this.
- otherwise, we should support multiple listen(). To do that, I recommend the use of rxjs. Using some Rx.Subject to broadcast the events.